src/Microsoft.Azure.SignalR/EndpointProvider/ConnectionStringParser.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal static class ConnectionStringParser
+    {
+        private const string EndpointProperty = "endpoint";
+        private const string AccessKeyProperty = "accesskey";
+        private const string VersionProperty = "version";
+        private const string PortProperty = "port";
+        // For SDK 1.x, only support Azure SignalR Service 1.x
+        private const string SupportedVersion = "1";
+        private const string ValidVersionRegex = "^" + SupportedVersion + @"\.\d+(?:[\w-.]+)?$";
```

Comment:
Can't we just try parse the version and avoid using regex?

Reply:
Just copied over here. And it is just a one-time thing, so it shouldn't be a big issue?